webpack/components/extensions/Hosts/BulkActions/BulkErrataWizard/BulkErrataWizard.js

```diff
@@ -0,0 +1,158 @@
+
+const BulkErrataWizard = () => {
+  const { modalOpen, setModalClosed: closeModal } = useForemanModal({ id: 'bulk-errata-wizard' });
+  const { selectedCount: initialSelectedHostCount, fetchBulkParams }
+    = useContext(ForemanActionsBarContext);
+
+  const [shouldValidateStep2, setShouldValidateStep2] = useState(false);
+  const [shouldValidateStep3, setShouldValidateStep3] = useState(false);
+  const [finishButtonLoading, setFinishButtonLoading] = useState(false);
+  const [selectedRexOption, setSelectedRexOption] = useState(dropdownOptions[0]);
+  const finishButtonText = __('Review and Finish');
+  const replacementResponse = !modalOpen ? { response: {} } : false;
+  const initialSelectedHosts = fetchBulkParams();
+  const apiOptions = { key: 'BULK_HOST_ERRATA' };
+  const defaultParams = { included: { search: initialSelectedHosts } };
+  const hostsBulkSelect =
+    useErrataHostsBulkSelect({ initialSelectedHosts, modalIsOpen: modalOpen });
+
+  const errataResponse = useTableIndexAPIResponse({
+    replacementResponse, // don't fetch data if modal is closed
+    apiUrl: getErrataUrl(),
+    apiOptions,
+    defaultParams,
+    method: 'post',
+  });
+
+  const {
+    response: {
+      results: errataResults,
+      ...errataMetadata
+    },
+  } = errataResponse;
+
+  const { total, page, subtotal } = errataMetadata;
+
+  const errataBulkSelect = useBulkSelect({
+    results: errataResults,
+    metadata: { total, page, selectable: subtotal },
+    idColumn: 'errata_id',
+  });
+
+  // eslint-disable-next-line no-restricted-globals
+  const selectionIsValid = count => count > 0 || isNaN(count);
+  const errataResultsPresent = errataResults?.length > 0;
+  const errataSelectionIsValid =
+    selectionIsValid(errataBulkSelect.selectedCount);
+  const hostSelectionIsValid = selectionIsValid(hostsBulkSelect.hostsBulkSelect.selectedCount);
+  let step2Valid = shouldValidateStep2 ? errataSelectionIsValid : true;
+  if (!errataResultsPresent) step2Valid = false;
+  const step3Valid = shouldValidateStep3 ? hostSelectionIsValid : true;
+  const step4Valid = hostSelectionIsValid && errataSelectionIsValid;
+
+  const BulkErrataWizardContextData = {
+    finishButtonText,
+    initialSelectedHostCount,
+    setShouldValidateStep2,
+    finishButtonLoading,
+    setFinishButtonLoading,
+    selectedRexOption,
+    setSelectedRexOption,
+    closeModal,
+    errataBulkSelect,
+    errataResults,
+    errataMetadata,
+    errataResponse,
+    hostsBulkSelect: hostsBulkSelect.hostsBulkSelect,
+  };
+  return (
+    <BulkErrataWizardContext.Provider value={BulkErrataWizardContextData}>
+      <Wizard
+        title="Manage errata wizard"
+        header={<WizardHeader title={__('Manage errata')} onClose={closeModal} />}
+      >
+        <WizardStep
+          name={__('Select errata')}
+          id="mew-step-2"
+          footer={{ isNextDisabled: !step2Valid, onClose: closeModal }}
+          status={step2Valid ? 'default' : 'error'}
```

Comment:
This shouldn't be error status while errata are loading.

Reply:
updated